Makefile

```
@@ -14,42 +26,18 @@ CONTAINER_RUNTIME=docker
 endif
 CONTAINER_RUNNABLE ?= $(shell $(CONTAINER_RUNTIME) -v > /dev/null 2>&1; echo $$?)
 
-.PHONY: unit_clean
-unit_clean: ## clean up the unit test artifacts created
-ifeq ($(CONTAINER_RUNNABLE), 0)
-		$(CONTAINER_RUNTIME) system prune -f
-endif
-		rm ${TEST_COVERAGE_FILE} ${TEST_COVERAGE_HTML_FILE} ${TEST_COVERAGE_FUNC_FILE} > /dev/null 2>&1
-
-.PHONY: unit
-unit: ## Run unit tests against code.
-ifeq ($(CONTAINER_RUNNABLE), 0)
-		$(CONTAINER_RUNTIME) run -it -v ${PWD}:/go/src -w /go/src docker.io/library/golang:${GO_VERSION}-alpine3.17 \
-         sh -e ./run-for-all-go-modules.sh "go test ./... -v -coverprofile ${TEST_COVERAGE_FILE}; \
-         go tool cover -html=${TEST_COVERAGE_FILE} -o ${TEST_COVERAGE_HTML_FILE}; \
-         go tool cover -func=${TEST_COVERAGE_FILE} -o ${TEST_COVERAGE_FUNC_FILE}"
-else
-		sh -e ./run-for-all-go-modules.sh  "\
-		 go test ./... -v -coverprofile ${TEST_COVERAGE_FILE} ; \
-		 go tool cover -html=${TEST_COVERAGE_FILE} -o ${TEST_COVERAGE_HTML_FILE} ; \
-		 go tool cover -func=${TEST_COVERAGE_FILE} -o ${TEST_COVERAGE_FUNC_FILE}"
-endif
+export CONTAINER_RUNTIME CONTAINER_RUNNABLE
 
-# Install link at https://golangci-lint.run/usage/install/ if not running inside a container
-.PHONY: lint
-lint: ## Run lint  against code.
-ifeq ($(CONTAINER_RUNNABLE), 0)
-		$(CONTAINER_RUNTIME) run -it -v ${PWD}:/go/src -w /go/src docker.io/golangci/golangci-lint:${GOLANG_CI_VER}-alpine \
-		 sh -e ./run-for-all-go-modules.sh golangci-lint run ./... -v --timeout 10m
-else
-		sh -e ./run-for-all-go-modules.sh golangci-lint run ./... -v --timeout 10m
-endif
+# find all subdirectories with a go.mod file in them
+GO_MOD_DIRS = $(shell find . -name 'go.mod' -exec sh -c 'echo \"$$(dirname "{}")\" ' \; )
+# NOTE: the above line is complicated due to the limited capabilities of busybox's `find`.
+# It meant to be equivalent with this:  find . -name 'go.mod' -printf "'%h' " 
+
+
+.PHONY: unit lint gosec unit_clean test
+# delegate these commands to the Makefiles next to the go.mod files
+unit lint gosec unit_clean test: 
+	for dir in $(GO_MOD_DIRS); do \
+		$(MAKE) -C "$$dir" $@ ; \
+	done
```

Shouldn't be better to include the find instruction here? The only advantage could be to avoid some executions in other targets.
```suggestion
	for dir in $(shell find . -name 'go.mod' -exec sh -c 'echo \"$$(dirname "{}")\" ' \;); do \
		$(MAKE) -C "$$dir" $@ ; \
	done
```

---

I find it more readable as it is. Wouldn't you agree?